### PR TITLE
Ft view cart 187570884 Buyer should be able to view their cart

### PR DESCRIPTION
**Pull request by kirengamartial**
### What does this PR do?

This PR implements the functionality for the buyer to view their cart. It retrieves the cart data from the buyer's session, fetches the product details from the database for each item in the cart, calculates the cart total, and sends the cart data to the frontend to be displayed to the buyer.
### Description of Task to be completed?

The task is to enable buyers to view their shopping cart, including the items they have added, the product details (name, price, image), and the total cost of the items in the cart.

### How should this be manually tested?

- Start the application and log in as a buyer.
- Add some products to the cart.
- Navigate to the cart route
- Verify that the cart displays all the items you added, along with their respective product details (name, price, image).
- Ensure that the cart total is correctly calculated and displayed.

### What are the relevant pivotal tracker stories
  https://www.pivotaltracker.com/n/projects/2703469

### CODE REVIEW

File: src/controllers/seller controller/sellerController.ts [hunk 1]
```diff
@@ -52,99 +50,111 @@ interface PaginationResponse {
 }
 
 class SellerController {
-async getAllProductsBySeller(req: Request, res: Response): Promise<void> {
```
Comment (Ishimwe7) [inline src/controllers/seller controller/sellerController.ts:55]: it's good that you have removed unnecessary codes

File: src/controllers/seller controller/sellerController.ts [hunk 2]
```diff
@@ -52,99 +50,111 @@ interface PaginationResponse {
...
-      };
-    }
-    
-    if (page > 1) {
-      response.previous = {
-        page: page - 1,
-        limit,
+
+    try {
+      const decodedToken = jwt.verify(token, process.env.VERIFICATION_JWT_SECRET as string) as JwtPayload;
```
Comment (Ishimwe7) [inline src/controllers/seller controller/sellerController.ts:62]: I think this is also un necessary as you are using a middleware before accessing this route

**Review by Ishimwe7 — CHANGES_REQUESTED** [review]
Consider feedbacks, otherwise you did a good job

**Review by robsdagreat — APPROVED** [review]
You work looks good @kirengamartial . We're good to go !

**Review by derrick-nuby — CHANGES_REQUESTED** [review]
please fix conficts

**Comment by kirengamartial** [comment]
> please fix conficts

it's done

**Review by Ishimwe7 — APPROVED** [review]
Thank you for considering requested changes

**Review by Balinda21 — APPROVED** [review]
Well done!

**Review by bahati10 — APPROVED** [review]
After the suggested changes now everything is good to go.

**Review by derrick-nuby — APPROVED** [review]
okay since you fixed errors everything is fine

**Pull request merged**